.github/workflows/main.yml

```diff
@@ -138,6 +138,67 @@ jobs:
           --runtime ${{ matrix.rid }}
           --self-contained
 
+      - name: Set permissions
+        if: ${{ !startsWith(matrix.rid, 'win-') }}
+        run: chmod +x ${{ matrix.app }}/bin/publish/${{ matrix.asset }}
+
+      - name: Generate macOS .app bundle resources
+        if: matrix.app == 'DiscordChatExporter.Gui' && startsWith(matrix.rid, 'osx-')
+        run: |
+          APP_NAME="${{ matrix.asset }}.app"
+          STAGING_DIR="app-bundle-staging"
+          APP_DIR="$STAGING_DIR/$APP_NAME"
+          CONTENTS_DIR="$APP_DIR/Contents"
+          MACOS_DIR="$CONTENTS_DIR/MacOS"
+          RESOURCES_DIR="$CONTENTS_DIR/Resources"
+          PUBLISH_DIR="${{ matrix.app }}/bin/publish"
+
+          # Create the macOS .app bundle directory structure
+          mkdir -p "$MACOS_DIR" "$RESOURCES_DIR"
+
+          # Move icon into the .app's Resources folder
+          mv "favicon.icns" "$RESOURCES_DIR/AppIcon.icns"
+
+          # Generate Info.plist metadata file with app information
+          cat > $CONTENTS_DIR/Info.plist << EOF
+            <?xml version="1.0" encoding="UTF-8"?>
+            <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+            <plist version="1.0">
+              <dict>
+                <key>CFBundleDisplayName</key>
+                <string>${{ matrix.asset }}</string>
+                <key>CFBundleName</key>
+                <string>${{ matrix.asset }}</string>
+                <key>CFBundleExecutable</key>
+                <string>${{ matrix.asset }}</string>
+                <key>NSHumanReadableCopyright</key>
+                <string>© Oleksii Holub</string>
+                <key>CFBundleIdentifier</key>
+                <string>com.Tyrrrz.${{ matrix.asset }}</string>
+                <key>CFBundleSpokenName</key>
+                <string>Discord Chat Exporter</string>
+                <key>CFBundleIconFile</key>
+                <string>AppIcon</string>
+                <key>CFBundleIconName</key>
+                <string>AppIcon</string>
+                <key>CFBundleVersion</key>
+                <string>${{ github.sha }}</string>
+                <key>CFBundleShortVersionString</key>
+                <string>${{ github.ref_type == 'tag' && github.ref_name || '999.9.9-ci' }}</string>
+                <key>NSHighResolutionCapable</key>
+                <true/>
+                <key>CFBundlePackageType</key>
+                <string>APPL</string>
+              </dict>
+            </plist>
+          EOF
+
+          # Copy all built application files into the .app's MacOS directory
+          find "$PUBLISH_DIR" -maxdepth 1 -mindepth 1 -exec mv "{}" "$MACOS_DIR/" \;
+
+          # Move the final .app bundle into the publish directory for upload
+          mv "$APP_DIR" "$PUBLISH_DIR"
```

Comment:
Can we move this to a parametrized script file instead? Ideally `pwsh` because that's x-platform, but `bash` is fine too.

Reply:
Done